docs/iris/src/IEP/IEP001.adoc

```diff
@@ -0,0 +1,193 @@
+
+Use a single helper for index by position, and a single helper for index by value. Helper names taken from pandas, but their behaviour is extended by making them callable to support named coordinates.
+
+|===
+.2+| 2+h|Index by
+h|Position h|Value
+
+h|Implicit dimension
+
+a|[source,python]
+----
+cube[:, 2]  # No change
+cube.iloc[:, 2]
+----
+
+a|[source,python]
+----
+cube.loc[:, 1.5]
+----
+
+h|Coordinate name
+
+a|[source,python]
+----
+cube[dict(height=2)]
+cube.iloc[dict(height=2)]
+cube.iloc(height=2)
+----
+
+a|[source,python]
+----
+cube.loc[dict(height=1.5)]
+cube.loc(height=1.5)
+----
+|===
+
+#### xarray style
+
+xarray introduces a second set of helpers for accessing named dimensions that provide the callable syntax `(foo=...)`.
+
+|===
+.2+| 2+h|Index by
+h|Position h|Value
+
+h|Implicit dimension
+
+a|[source,python]
+----
+cube[:, 2]  # No change
+----
+
+a|[source,python]
+----
+cube.loc[:, 1.5]
+----
+
+h|Coordinate name
+
+a|[source,python]
+----
+ cube[dict(height=2)]
+ cube.isel(height=2)
+----
+
+a|[source,python]
+----
+cube.loc[dict(height=1.5)]
+cube.sel(height=1.5)
+----
+|===
+
+### Slices
+
+The semantics of position-based slices will continue to match that of normal Python slices. The start position is included, the end position is excluded.
+
+Value-based slices will be stricly inclusive, with both the start and end values included. This behaviour differs from normal Python slices but is in common with pandas.
+
+Just as for normal Python slices, we do not need to provide the ability to control the include/exclude behaviour for slicing.
+
+### Value-based indexing
+
+#### Equality
+
+Should the behaviour of value-based equality depend on the data type of the coordinate?
+
+* integer: exact match
+* float: tolerance match, tolerance determined by bit-width
+* string: exact match
+
+#### Scalar/category
+
+If/how to deal with category selection `cube.loc(season='JJA')`? Defer to `groupby()`?
+
+`cube.loc[12]` - must always match a single value or raise KeyError, corresponding dimension will be removed
+`cube.loc[[12]]` - may match any number of values?  (incl. zero?), dimension will be retained
+
+### Out of scope
+
+* Deliberately enhancing the performance.
+This is a very valuable topic and should be addressed by subsequent efforts.
+
+* Time/date values as strings.
+Providing pandas-style string representations for convenient representation of partial date/times should be addressed in a subsequent effort - perhaps in conjunction with an explicit performance test suite.
+There is a risk that this topic could bog down when dealing with non-standard calendars and climatological date ranges.
+
+## Work required
+
+* Implementations for each of the new helper objects.
+* An update to the documentation to demonstrate best practice. Known impacted areas include:
+** The "Subsetting a Cube" chapter of the user guide.
+
+### TODO
+* Multi-dimensional coordinates
+* Non-orthogonal coordinates
+* Bounds
```

Comment:
this is an interesting facet.  I'd like to think about this more.  find sample point within bounds may be an interesting way to handle float matching in some cases.  opportunities and dragons, I suspect